worker-versioning/src/example.ts

```diff
@@ -0,0 +1,120 @@
+import { Connection, Client } from '@temporalio/client';
+import { uuid4 } from '@temporalio/workflow';
+import { proceeder, versioningExample } from './workflowsV1';
+import { Worker } from '@temporalio/worker';
+import * as activities from './activities';
+
+async function run() {
+  const connection = await Connection.connect();
+  const client = new Client({
+    connection,
+  });
+  const taskQueue = 'versioned-queue_' + uuid4();
+
+  // First, let's make the task queue use the build id versioning feature by adding an initial
+  // default version to the queue:
+  await client.taskQueue.updateBuildIdCompatibility(taskQueue, {
+    operation: 'addNewIdInNewDefaultSet',
+    buildId: '1.0',
+  });
+
+  // Start a 1.0 worker
+  const worker1 = await Worker.create({
+    workflowsPath: require.resolve('./workflowsV1'),
+    activities,
+    taskQueue,
+    buildId: '1.0',
+    useVersioning: true,
+  });
+  const worker1Run = worker1.run();
+
+  // Start a workflow which will run on the 1.0 worker
+  const firstWorkflowID = 'worker-versioning-first_' + uuid4();
+  const firstWorkflow = await client.workflow.start(versioningExample, {
+    workflowId: firstWorkflowID,
+    taskQueue,
+    workflowExecutionTimeout: '5 minutes',
+  });
+
+  // Signal the workflow to drive it
+  await firstWorkflow.signal(proceeder, 'go');
+
+  // Give a chance for the worker to process the signal
+  await new Promise((resolve) => setTimeout(resolve, 3000));
+
+  // Add a new compatible version to the queue
+  await client.taskQueue.updateBuildIdCompatibility(taskQueue, {
+    operation: 'addNewCompatibleVersion',
+    buildId: '1.1',
+    existingCompatibleBuildId: '1.0',
+  });
+
+  // Stop the old worker, and start a 1.1 worker. We do this to speed along the example, since the
+  // 1.0 worker may continue to process tasks briefly after the version update.
+  worker1.shutdown();
+  await worker1Run;
+  const worker11 = await Worker.create({
```

Comment:
```suggestion
  const worker1dot1 = await Worker.create({
```

i read as worker eleven